bundles/org.openhab.binding.linuxinput/src/main/java/org/openhab/binding/linuxinput/internal/evdev4j/jnr/EvdevLibrary.java

```diff
@@ -0,0 +1,236 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.linuxinput.internal.evdev4j.jnr;
+
+import static org.openhab.binding.linuxinput.internal.evdev4j.Utils.constantFromInt;
+import static org.openhab.binding.linuxinput.internal.evdev4j.jnr.Utils.trimEnd;
+
+import java.util.Optional;
+
+import org.openhab.binding.linuxinput.internal.evdev4j.Utils;
+
+import jnr.constants.Constant;
+import jnr.ffi.LibraryLoader;
+import jnr.ffi.Runtime;
+import jnr.ffi.Struct;
+import jnr.ffi.annotations.IgnoreError;
+import jnr.ffi.annotations.In;
+import jnr.ffi.annotations.Out;
+import jnr.ffi.byref.PointerByReference;
+import jnr.ffi.mapper.FunctionMapper;
+
+/**
+ * JNR library for libdevdev library (libevdev.h).
+ *
+ * @author Thomas Weißschuh - Initial contribution
+ */
+@IgnoreError
+public interface EvdevLibrary {
+    static EvdevLibrary load() {
+        FunctionMapper evdevFunctionMapper = (functionName, context) -> "libevdev_" + trimEnd("_", functionName);
+        return LibraryLoader.create(EvdevLibrary.class).library("evdev").mapper(evdevFunctionMapper).load();
+    }
+
+    static Handle makeHandle(EvdevLibrary lib) {
+        return new Handle(Runtime.getRuntime(lib));
+    }
+
+    enum GrabMode {
+        GRAB(3),
+        UNGRAB(4);
+
+        private int value;
+
+        GrabMode(int value) {
+            this.value = value;
+        }
+
+        public int getValue() {
+            return value;
+        }
+    }
+
+    final class InputEvent extends Struct {
+        public UnsignedLong sec = new UnsignedLong();
+        public UnsignedLong usec = new UnsignedLong();
+        public Unsigned16 type = new Unsigned16();
+        public Unsigned16 code = new Unsigned16();
+        public Signed32 value = new Signed32();
+
+        public InputEvent(Runtime runtime) {
+            super(runtime);
+        }
+    }
+
+    final class Handle extends Struct {
+        public Handle(Runtime runtime) {
+            super(runtime);
+        }
+    }
+
+    int new_from_fd(int fd, @Out PointerByReference handle);
```

Comment:
In theory yes. By default the name has to match the name of the C function exactly. It is possible to provide an adapter that translates the name.
In this case here I strongly prefer the name matching with the C function.
This will make it easier to correlate these functions with the names in the documentation